Issue #14: Added Echo Request Handler for adding Meds.

jking31cs opened the pull request:
Added a simple MedService/MockMedService that adds a medicine by its
name.

Successfully grabbed access token from user object passed into the
Session object for Alexa, and grabbed user information from it.

Added MockAuthUserService that validates against google with the
access token and creates a fake AuthUser with that information.

Moved all Echo specific code into its own package, added postman
tests, and testing the rest endpoints to ensure that they work.

This does successfully use OAuth for the Echo, but
not for the Rest Endpoints.

jking31cs committed:
Issue #14: Added Echo Request Handler for adding Meds.

Added a simple MedService/MockMedService that adds a medicine by its
name.

Successfully grabbed access token from user object passed into the
Session object for Alexa, and grabbed user information from it.

Added MockAuthUserService that validates against google with the
access token and creates a fake AuthUser with that information.

Moved all Echo specific code into its own package, added postman
tests, and testing the rest endpoints to ensure that they work.

This does successfully use OAuth for the Echo, but
not for the Rest Endpoints.

jking31cs commented on server/src/test/java/info/doseamigos/meds/MedTest.java:
```diff
@@ -56,10 +56,10 @@ public void testGetName() throws Exception {
-    public void testSetName() throws Exception {
-        new Med().setName(null);
-    }
+//    @Test(expectedExceptions = NullPointerException.class)
```
There was a weird issue where Alexa was doing the wrong intent, and kept causing NPE's, so I made this nullable and commented out the test.  I figured out that the issue was that a trailing space in testing caused it to think it was a different intent.  I might be able to undo this tomorrow once we're in a better more stable state.

jking31cs commented on server/src/main/java/info/doseamigos/authusers/MockAuthUserService.java:
```diff
@@ -0,0 +1,58 @@
+public class MockAuthUserService implements AuthUserService {
+
+    private static final Logger log = LoggerFactory.getLogger(MockAuthUserService.class);
+    @Override
```
Need a newline before this for consistency.

jking31cs commented on server/src/main/java/info/doseamigos/authusers/DefaultAuthUserService.java:
```diff
@@ -0,0 +1,79 @@
+/**
+ * Implementation of AuthUserService using Google OAuth URLs.
+ */
+public class DefaultAuthUserService implements AuthUserService {
```
The code works, but only locally.  With Amazon Lambda, it cannot connect to the DB.  I've had all sorts of headaches about this, and this isn't being used for the demo tomorrow.

jking31cs commented on server/src/main/java/info/doseamigos/authusers/AuthUserGuiceModule.java:
```diff
@@ -0,0 +1,14 @@
+import com.google.inject.AbstractModule;
+
+/**
+ * Created by jking31cs on 6/27/16.
```
damnit, need to remove default comments :(

jking31cs commented on server/src/main/java/info/doseamigos/authusers/AuthUser.java:
```diff
@@ -5,22 +5,25 @@
 public class AuthUser {
 
-    private long authUserId;
+    private Long authUserId;
```
Using a Long for new AuthUser creation.

jking31cs commented on server/src/main/java/info/doseamigos/authusers/AuthUser.java:
```diff
@@ -5,22 +5,25 @@
+    private Long authUserId;
     private AmigoUser amigoUser;
     private String email;
+    private BigInteger googleRef;
```
Google's id is too big for a long, so I'm using BigInteger, and for the DB, I'm storing the string representation.

mckoon replied:
Is it ever used as a number?  If not, why not just a String always?

jking31cs replied:
... good point

jking31cs committed:
Issue #14, Issue #5, Issue #6: Fixes not for demo.

This is a separate commit of fixing stuff up so that we can
revert back to demoable state for worst case scenario.

jking31cs commented:
Made the fixes in a separate repo since the demo releases are from the previous commit.  I didn't want to clobber that commit for that reason.

mckoon commented:
> Made the fixes in a separate repo

You mean "commits"?

jking31cs commented:
I did mean commits, I'm tired...

mckoon commented on server/src/main/java/info/doseamigos/authusers/AuthUser.java:
```diff
@@ -5,22 +5,25 @@
 import org.apache.commons.lang3.builder.HashCodeBuilder;
 import org.apache.commons.lang3.builder.ToStringBuilder;
 
+import java.math.BigInteger;
```
Unused?

mckoon commented on server/src/main/java/info/doseamigos/authusers/AuthUserDao.java:
```diff
@@ -0,0 +1,38 @@
+package info.doseamigos.authusers;
+
+import java.math.BigInteger;
```
Unused?

mckoon commented on server/src/main/java/info/doseamigos/authusers/MockAuthUserService.java:
```diff
@@ -0,0 +1,59 @@
+package info.doseamigos.authusers;
+
+import java.io.IOException;
+import java.math.BigInteger;
```
Unused?

jking31cs commented:
Cleaned up imports

mckoon commented:
+1
